addons/website/models/ir_ui_view.py

```diff
@@ -40,8 +40,9 @@ def _get_pwd(self):
     def _set_pwd(self):
         crypt_context = self.env.user._crypt_context()
         for r in self:
-            r.sudo().visibility_password = crypt_context.encrypt(r.visibility_password_display)
-            r.visibility = r.visibility  # double check access
+            if r.type == 'qweb' and r.visibility_password_display:  # empty if no password and not updated
+                r.sudo().visibility_password = crypt_context.encrypt(r.visibility_password_display)
```

Comment:
If visibility_password_display is empty here doesn't mean this is the result of explicitly trying to empty it? Should it really just be ignored?